src/main/java/de/terrestris/momo/service/MomoApplicationService.java

```diff
@@ -0,0 +1,279 @@
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.ApplicationContext;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Service;
+
+import de.terrestris.momo.dto.ApplicationData;
+import de.terrestris.shogun2.dao.ApplicationDao;
+import de.terrestris.shogun2.dao.ExtentDao;
+import de.terrestris.shogun2.dao.LayerDao;
+import de.terrestris.shogun2.dao.LayoutDao;
+import de.terrestris.shogun2.dao.MapConfigDao;
+import de.terrestris.shogun2.dao.MapControlDao;
+import de.terrestris.shogun2.dao.MapDao;
+import de.terrestris.shogun2.dao.ModuleDao;
+import de.terrestris.shogun2.dao.UserDao;
+import de.terrestris.shogun2.model.Application;
+import de.terrestris.shogun2.model.User;
+import de.terrestris.shogun2.model.layer.Layer;
+import de.terrestris.shogun2.model.layer.util.Extent;
+import de.terrestris.shogun2.model.layout.Layout;
+import de.terrestris.shogun2.model.map.MapConfig;
+import de.terrestris.shogun2.model.map.MapControl;
+import de.terrestris.shogun2.model.module.CompositeModule;
+import de.terrestris.shogun2.model.module.Map;
+import de.terrestris.shogun2.model.module.Module;
+import de.terrestris.shogun2.model.security.Permission;
+import de.terrestris.shogun2.service.ApplicationService;
+import de.terrestris.shogun2.service.ExtentService;
+import de.terrestris.shogun2.service.LayerService;
+import de.terrestris.shogun2.service.LayoutService;
+import de.terrestris.shogun2.service.MapConfigService;
+import de.terrestris.shogun2.service.MapControlService;
+import de.terrestris.shogun2.service.MapService;
+import de.terrestris.shogun2.service.ModuleService;
+import de.terrestris.shogun2.service.UserService;
+
+/**
+ *
+ * @author Nils Bühner
+ * @see ApplicationService
+ *
+ */
+@Service("momoApplicationService")
+public class MomoApplicationService<E extends Application, D extends ApplicationDao<E>>
+		extends ApplicationService<E, D> {
+
+	private static final String BEAN_ID_DEFAULT_MAP = "defaultMap";
+
+	private static final String BEAN_ID_DEFAULT_MAP_CONTAINER = "defaultMapContainer";
+
+	private static final String BEAN_ID_DEFAULT_MAX_EXTENT = "defaultMaxExtent";
+
+	private static final String BEAN_ID_DEFAULT_RESOLUTIONS = "defaultResolutions";
+
+	private static final String BEAN_ID_MAX_RES = "res19";
+
+	private static final String BEAN_ID_MIN_RES = "res01";
+
+	private static final Set<String> DEFAULT_MAP_CONTROLS = Collections.unmodifiableSet(
+			new HashSet<String>(Arrays.asList("Attribution", "Zoom", "Rotate", "ScaleLine", "ZoomSlider")));
+
+	private static final List<String> DEFAULT_MAP_CONTAINER_MODULES = Collections.unmodifiableList(Arrays
+			.asList("Show meta toolbar button", "Meta info toolbar for the map", "Tools Toolbar", "Gazetteer Grid"));
+
+	private static final List<String> DEFAULT_VIEWPORT_MODULES = Collections
+			.unmodifiableList(Arrays.asList("Viewport Header", "Legend Tree", "Gazetteer Grid"));
+
```

Comment:
Hmmm, it's probably not the best approach to use here the hardcoded module names as they can be modified/deleted/renamed.. 

But as we don't have anything else at the moment you can leave it firstly as is.. Just keep it in mind, that could be a problem lately..